### ISSUE TITLE
(Beta 2.0.6 and LOOT 0.9.0) Unable to Start LOOT through MO2

### ISSUE THREAD

**Issue by Kesnit** [TanninOne/modorganizer#803]
I've seen several threads on this issue and have tried to fixes they list. However, I cannot get LOOT to run through MO. I am getting the error message 
failed to spawn "LOOT.exe" : failed to start process (The parameter is incorrect. [87]). I get the error when I click on the Sort button and when I try to run LOOT through MO2. (LOOT runs fine through MO1.) 

Windows 7
MO 2.0.6 (Installed in c:\Downloads\Fallout 4\Mod Organizer)
LOOT 0.9 (installed in c:\Games\Loot FO4)

In the Modify Executables, I've tried the arguments 
--game=Fallout4
--single-parameter
together and separately. 

I downloaded LOOT.API and extracted it into the loot folder under MO. 

I'm sure there is a simple solution, but I haven't been able to find it.

**Comment by GTSlayer** [comment]
Install both the 32 and 64 bit versions of vcredist. You'll find copies in the Fallout 4 folder under the folder _CommonRedist. 

Uninstall whatever version you have currently. Then install 32 bit, then 64 bit. 

If you do the opposite it likely won't allow you to install the 32 bit. This fixed all loot issues for me, both with the sort button and the application itself when added to MO.

**Comment by dt78** [comment]
@GTSlayer Thank you! That finally fixed my issue trying to get Loot to work in MO v2.0.6. I had the program launching before. However once it was running, it didn't see anything in the MO virtual data. Just vanilla and DLC esm's would show.

@Kesnit If Loot is not launching try downloading the newer usvfs DLL files from the [releases page](https://github.com/TanninOne/modorganizer/releases). Download and extract the contents of usvsf.7z into the MO program folder. Overwrite the previous version of the DLLs or back them up first (up to you). If MO can now launch Loot, and it sees and sorts you plugins, great. You're done. If loot launches, but doesn't show your full load list, follow GTSlayer's instructions to reinstall vcredest. 

My path to MO2/Loot success:

1. MO v2.0.6
2. usvsf.7z
3. LOOT 0.9
4. Reinstall vcredist (Fallout4\_CommonRedist)

**Comment by Kesnit** [comment]
I tried both of the above, but neither worked.

One thing I should have pointed out (but it didn't register with me last night) is when I try to start LOOT using the Sort button, I get the error message
failed to spawn "lootcli.exe" : failed to start process (The parameter is incorrect. [87]).

The newest version of vcredist (2015) did not work. (I got an error that I had a missing .dll when I tried to run MO2.) I went back to 2013 and MO2 runs now. That may be the issue with running LOOT.

The only file in _CommonRedist was installscript.vdf. I had to download vcredist from the Web.

**Comment by dt78** [comment]
@Kesnit the version of vcredest should be 2012. That is the version that should have been sitting in the _CommonRedist folder. And make sure you install the x86 then the x64 version. You need both.

BTW in win7+ you can (or at least your supposed to) have multiple versions of libraries like these without causing issues. You can install 2012, 2013, and 2015 (32bit and 64) all on the same system. You might have other things on your PC that require the other versions, so don't rip them out unless you are sure you don't need them.

**Comment by amalys11** [comment]
I'm having similar problems running loot through MO2 (both running the executable as well as using the sort button built into MO2) for Fallout 4. It worked though MO2 on 6/5, but when I tried running Loot on 6/12 this issue started. Loot works outside of MO2 without issue.

I've tried everything in the various threads about similar issues and nothing has helped. I've installed the latest versions of MO2 and LOOT. I've reinstalled all the vcredists (2012, 2013, and 2015) multiple times. Made sure I'm using the latest usvsf.7z files.

When running the executable though MO2: loot opens, the loading bar gets to the "Parsing, Merging, and Evaluating Metadata" step, then a windows error message pops up stating that LOOT.exe has stopped working.
When using the sort button in MO2: I get and error message stating "loot failed. Exit code was: 3221225477"

**Comment by TanninOne** [comment]
Please attach a usvfs_<date>.log from such a run

**Comment by amalys11** [comment]
Sure thing. I opened MO2, tried running the executable, then tried using the sort button. Here's the log

[usvfs-2016-06-14_22-41-12.log.txt](https://github.com/TanninOne/modorganizer/files/315091/usvfs-2016-06-14_22-41-12.log.txt)

**Comment by amalys11** [comment]
I resolved this. I wasn't aware of that log file, but when I looked for the exception it lists at the end, I found this previous issue https://github.com/TanninOne/modorganizer/issues/567

Turns out I had a random ba2 file in the data folder (a hack I did from months ago), and it belonged to a mod I recently updated, which I'm guessing caused a conflict since the same file existed in the actual folder AND a MO folder

**Comment by Kesnit** [comment]
I have no log files. I did a search for "usvf", but only have usvf_x86.dll, usvf_64.dll and usvf_proxy.exe 

My log folder is empty. MO2 is set to Debug log level.

**Comment by drbeavi5** [comment]
I've got a similar issue. I get the error message 

> Error code: -1; Failed to get game data. Details: boost::filesystem::directory_iterator::construct: The system cannot find the path specified: "E:\Steam2\SteamApps\common\Fallout 4\Data"

[usvfs-2016-06-15_19-16-19.txt](https://github.com/TanninOne/modorganizer/files/317016/usvfs-2016-06-15_19-16-19.txt)

**Comment by Cladors** [comment]
I was unable to start loot also with the MO2 version being the same and LOOT being Version 0.9.1.
F4SE works fine though
I have an unrecognised exception occurring though

Unknown exception13:28:27.161 [E] error: (unknown error [193])

[usvfs-2016-07-13_12-24-14.log.txt](https://github.com/TanninOne/modorganizer/files/361471/usvfs-2016-07-13_12-24-14.log.txt)

**Comment by jpbruckler** [comment]
Yesterday MO2 was working perfectly. Today I was getting this same error when launching LOOT. The only thing that changed between yesterday and today was that I installed the Creation Kit, which installed VC++ 2015 x86.

Since I already had the x64 version installed prior to the CCK installing the x86 version, I figured the best thing to do was to remove and reinstall the Visual C++ redists on my machine. I actually just removed and reinstalled all of them.

After reinstalling and a reboot, MO2 is working perfectly again.

Here's a link to a page that will let you download a script that will download all the VC++ redistributables: https://deploymentbunny.com/2014/08/05/powershell-is-king-download-all-vc-runtimes-using-a-script/

Assuming that you downloaded them all using the above script, I wrote a PowerShell script for installing them in order, at [https://gist.github.com/jpbruckler/61a757e13e66fa1071a21402af678545](url)

**Comment by Bohauo** [comment]
I had a problem with LOOT, and all 3rd party programs linked in MO2
I got Error code 87...
I don't know how or what happened with loot.exe, but somehow i changed its own permissions, so i as system-admin couldn't do anything with loot.exe. My fix was to restart my computer in a fail-safe state and delete the file, restart to normal state, download, and reinstall loot to the same location.

All my problems vanished with that.

**Comment by Pathos23** [comment]
Had the same issue as the one stated above: "failed to spawn "lootcli.exe" : failed to start process (The parameter is incorrect. [87])."

Using MO 2.0.8.3b & LOOT 0.11.0

@jpbruckler solution solved the issue for me.

**Comment by Ruftinator** [comment]
After a bit of fiddling around i found some stuff that helped me. On the Skyrim mods subreddit (i don't know if that's correct as i don't use Reddit for too long) i found a thread in which a person under the name "arcline111" suggested a reinstall of MO2, which i did (ofcourse after backing up my "mods" and "profile's" folder). When i had MO2 up and running again i redownloaded LOOT and installed that too. I found out that it had worked, LOOT was up and running, my mods were all showing and even though i haven't tried it as of yet i think it'll work just like it should. Anyway, back to the point. After closing LOOT Norton had a pop-up message telling me it had put a program called "usvs_proxy.exe" in quarantine. After a reboot (Norton protocol i guess) i imediately checked to see if LOOT would still run. It didn't... better yet, it gave me the same error that this thread is about.

Long story short. The problem for me was on behalf of my anti-virus protection. I don't know how it works with other programs, but for me it was as easy as going into the safety history of Norton and then telling it to lift the quarantine and exclude usvs_proxy.exe from future scans.

I hope this was helpful in any kind of way :)

**Comment by czod** [comment]
Unbelievable!  I've been searching for this solution for MONTHS!  I'd even gone so far as to exclude the MO2 directory from Norton but I didn't realize that it had already quarantined usvs_proxy.exe.  Thank you so much Ruftinator.  Now I just have to see if I can still load my save game after running unsorted for months.

**Comment by Ruftinator** [comment]
@czod
Glad i could help :)

**Comment by lordcuzzins** [comment]
@Ruftinator Let me echo @czod sentiments.  I have been researching this topic non-stop for a while and checking every board I could think of to see if anyone had a solution.  Tried everything with no luck.  Never considered Norton to be the culprit!  Thank you.

**Comment by Ruftinator** [comment]
@lordcuzzins thanks for the reply man, it may sound weird, but i’m so damn happy for your reply :)

Thanks for making my day
